library/gen_837i.inc.php

```diff
@@ -0,0 +1,1946 @@
+* Please help the overall project by sending changes you make to the authors and to the LibreHealth EHR community.
+*
+*/
+
+$fake_register_globals=false;
+$sanitize_all_escapes=true;
+
+require_once("Claim.class.php");
+
+
+function gen_837I($pid, $encounter, &$log, $encounter_claim=false) {
+
+  $today = time();
+  $out = '';
+  $claim = new Claim($pid, $encounter);
+  $edicount = 0;
+
+  // This is true for the 5010 standard, false for 4010.
+  // x12gsversionstring() should be "005010X222A1" or "004010X098A1".
+  $CMS_5010 = strpos($claim->x12gsversionstring(), '5010') !== false;
+
+  $log .= "Generating claim $pid-$encounter for " .
+    $claim->patientFirstName()  . ' ' .
+    $claim->patientMiddleName() . ' ' .
+    $claim->patientLastName()   . ' on ' .
+    date('Y-m-d H:i', $today) . ".\n";
+
+
+
+  $out .= "ISA" .
+    "*" . $claim->x12gsisa01() .
+    "*" . $claim->x12gsisa02() .
+    "*" . $claim->x12gsisa03() .
+    "*" . $claim->x12gsisa04() .
+    "*" . $claim->x12gsisa05() .
+    "*" . $claim->x12gssenderid() .
+    "*" . $claim->x12gsisa07() .
+    "*" . $claim->x12gsreceiverid() .
+	// date of transmission "*030911" .
+	"*" . date('Ymd', $today) .
+    //Time of transmission "*1630" .
+	"*" . date('Hi', $today) .
+	"*" . ($CMS_5010 ? "^"     : "U"    ) .
+    "*" . ($CMS_5010 ? "00501" : "00401") .
+    "*000000001" .
+    "*" . $claim->x12gsisa14() .
+    "*" . $claim->x12gsisa15() .
+    "*:" .
+    "~\n";
+
+  $out .= "GS" .
+    "*HC" .
+    "*" . $claim->x12gsgs02() .
+    "*" . trim($claim->x12gs03()) .
+    "*" . date('Ymd', $today) .
+    "*" . date('Hi', $today) .
+    "*1" .
+    "*X" .
+   // "*" . $claim->x12gsversionstring() . 
+    "*" . "005010X223A2".
+    "~\n";
+
+  ++$edicount;
+  $out .= "ST" .
+    "*837" .
+    "*0021" .
+    //($CMS_5010 ? ("*" . $claim->x12gsversionstring()) : "") .
+	"*" . "005010X223A2".
+    "~\n";
+
+  ++$edicount;
+  $out .= "BHT" .
+    "*0019" .                             // 0019 is required here
+    "*00" .                               // 00 = original transmission
+    "*0123" .                             // reference identification
+    "*" . date('Ymd', $today) .           // transaction creation date
+    "*" . date('Hi', $today) .            // transaction creation time
+    ($encounter_claim ? "*RP" : "*CH") .  // RP = reporting, CH = chargeable
+    "~\n";
+
+  if (!$CMS_5010) {
+    // This segment was deleted for 5010.
+    ++$edicount;
+    $out .= "REF" .
+      "*87" .
+      "*" . $claim->x12gsversionstring() .
+      "~\n";
+  }
+
+  ++$edicount;
+  //Field length is limited to 35. See nucc dataset page 63 www.nucc.org
+  $billingFacilityName = substr($claim->billingFacilityName(), 0, $CMS_5010 ? 60 : 35);
+  $out .= "NM1" .       // Loop 1000A Submitter stays in the 837I
+    "*41" .
+    "*2" .
+    "*" . $billingFacilityName .
+    "*" .
+    "*" .
+    "*" .
+    "*" .
+    "*46";
+   if (trim($claim->x12gsreceiverid()) == '470819582') { // if ECLAIMS EDI
```

Comment:
Other exceptions needed here.  I suppose I should bounce this off of other stuff I have.